packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObjectFactory } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { SharedCounter } from "@fluidframework/counter";
+import { DataObjectWithCounter, dataObjectWithCounterFactory } from "./dataObjectWithCounter";
+
+export class RootDataObjectWithChildDataObject extends DataObjectWithCounter {
+    private unrefTimeStampMs?: number;
+    private unreferencedChild?: DataObjectWithCounter;
+    private child?: DataObjectWithCounter;
+    private get childKey(): string {
+        assert(this.context.clientId !== undefined, `client id needs to be defined to retrieve the child key!`);
+        return `childKey:${this.context.clientId}`;
+    }
+
+    public static get type(): string {
+        return "RootDataObjectWithChildDataObject";
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        await super.hasInitialized();
+        await this.createAndReferenceChild();
+        // TODO: Start logic goes here after we agree on what it should be.
+    }
+
+    public async unreferenceChild() {
```

Comment:
All these methods that are specific to this class implementation should be private. You can make them public as and when you use them. Right now, its hard to know which ones will be exposed outside of this class.

Reply:
Note: Leaving them public so the build passes